### PR TITLE
Update allowlist/denylist terminology

### LINKED ISSUE #1049: Update terminology: whitelist -> allowlist, blacklist -> denylist
Many companies have modified their terminology to be not only more inclusive, but more clear. Let's follow suit! Here are some rule messages that will need updating (probably not exhaustive, this is just from a quick search of the registry):

- [x] ruby.lang.security.model-attributes-attr-protected
- [x] ruby.lang.secuirty.model-attributes-attr-accessible
- [x] python.flask.security.injection.ssrf-requests
- [x] python.django.security.injection.ssrf.ssrf-injection-urllib
- [x] python.django.security.injection.ssrf.ssrf-injection-requests
- [x] java.lang.security.audit.unvalidated-redirect
- [x] javascript.angular.security.detect-angular-resource-loading
- [x] javascript.browser.security.dom-based-xss

I've also seen blocklist used instead of denylist so we can decide which one we like.

### PR DESCRIPTION
**Pull request by minusworld**
Closes https://github.com/returntocorp/semgrep-rules/issues/1049.

### CODE REVIEW

File: java/lang/security/audit/unvalidated-redirect.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ rules:
   message: |
     Application redirects to a destination URL specified by a user-supplied
     parameter that is not validated. This could direct users to malicious locations.
-    Consider using a whitelist to validate URLs.
+    Consider using a allowlist to validate URLs.
```
Comment (sabrinabrogren) [inline java/lang/security/audit/unvalidated-redirect.yaml]:
```suggestion
    Consider using an allowlist to validate URLs.
```

File: python/django/security/injection/ssrf/ssrf-injection-requests.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ rules:
   message: |
     Data from request object is passed to a new server-side request.
     This could lead to a server-side request forgery (SSRF). To mitigate,
-    ensure that schemes and hosts are validated against a whitelist,
+    ensure that schemes and hosts are validated against a allowlist,
```
Comment (sabrinabrogren) [inline python/django/security/injection/ssrf/ssrf-injection-requests.yaml]:
```suggestion
    ensure that schemes and hosts are validated against an allowlist,
```

File: python/flask/security/injection/ssrf-requests.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ rules:
   - python
   severity: ERROR
   message: Data from request object is passed to a new server-side request. This could lead to a server-side request forgery
-    (SSRF). To mitigate, ensure that schemes and hosts are validated against a whitelist, do not forward the response to the
+    (SSRF). To mitigate, ensure that schemes and hosts are validated against a allowlist, do not forward the response to the
```
Comment (sabrinabrogren) [inline python/flask/security/injection/ssrf-requests.yaml]:
```suggestion
    (SSRF). To mitigate, ensure that schemes and hosts are validated against an allowlist, do not forward the response to the
```

File: ruby/lang/security/mass-assignment-protection-disabled.yaml [hunk 4]
```diff
@@ -5,7 +5,7 @@ rules:
     Mass assignment protection disabled for '$MODEL'. This could
     permit assignment to sensitive model fields without intention. Instead,
     use 'attr_accessible' for the model or disable mass assigment using
-    'config.active_record.whitelist_attributes = true'.
+    'config.active_record.allowlist_attributes = true'.
```
Comment (sabrinabrogren) [inline ruby/lang/security/mass-assignment-protection-disabled.yaml]: Should this be changed? It looks like it's code so we'll need to make sure this variable name was changed too.

**Comment by sabrinabrogren** [comment]
Thanks so much for getting to this, @minusworld  🙏

File: ruby/lang/security/mass-assignment-protection-disabled.yaml (hunk 4 above)
Comment (colleend) [reply ruby/lang/security/mass-assignment-protection-disabled.yaml:8]: this is a configuration status in Ruby, (they still use whitelist/blacklist), so we should keep this 😅

Comment (minusworld) [reply ruby/lang/security/mass-assignment-protection-disabled.yaml]: Changed back, good catch!

**Review by sabrinabrogren — APPROVED** [review]

**Pull request merged**